lib/aws/card-vault/components.ts

```diff
@@ -0,0 +1,387 @@
+      {
+        modulusLength: 2048,
+      },
+    );
+
+    this.locker_pair.public_key = locker_public_key
+      .export({ type: "spki", format: "pem" })
+      .toString();
+    this.locker_pair.private_key = locker_private_key
+      .export({ type: "pkcs8", format: "pem" })
+      .toString();
+
+    const { tenant_private_key, tenant_public_key } = generateKeyPairSync(
+      "rsa",
+      {
+        modulusLength: 2048,
+      },
+    );
+
+    this.hyperswitch.public_key = tenant_public_key
+      .export({ type: "spki", format: "pem" })
+      .toString();
+    this.hyperswitch.private_key = tenant_private_key
+      .export({ type: "pkcs8", format: "pem" })
+      .toString();
+
+    let customData = readFileSync("lib/aws/card-vault/user-data.sh", "utf8")
+      .replaceAll("{{db_user}}", locker_data.database.user)
+      .replaceAll("{{kms_enc_db_pass}}", locker_data.database.password)
+      .replaceAll("{{db_host}}", locker_data.database.host)
+      .replaceAll("{{kms_enc_master_key}}", locker_data.master_key)
+      .replaceAll("{{kms_enc_lpriv_key}}", this.locker_pair.private_key)
+      .replaceAll("{{kms_enc_tpub_key}}", this.hyperswitch.public_key)
+      .replaceAll("{{kms_id}}", locker_data.kms.id)
+      .replaceAll("{{kms_region}}", locker_data.kms.region);
+
+    this.instance = new ec2.Instance(scope, "locker-ec2", {
+      instanceType: ec2.InstanceType.of(
+        ec2.InstanceClass.T3,
+        ec2.InstanceSize.MEDIUM,
+      ),
+      machineImage: new ec2.AmazonLinuxImage(),
+      vpc,
+      vpcSubnets: {
+        subnetGroupName: SubnetNames.IsolatedSubnet,
+      },
+      securityGroup: sg,
+      keyName: aws_key_pair.keyName,
+      userData: ec2.UserData.custom(customData),
+      role: locker_role,
+    });
+
+    new cdk.CfnOutput(scope, "Locker-ec2-IP", {
+      value: `http://${this.instance.instancePrivateIp}/health`,
+      description: "try health api",
+    });
+  }
+
+  addClient(sg: ec2.ISecurityGroup, port: ec2.Port) {
+    this.sg.addIngressRule(sg, port);
+    sg.addEgressRule(this.sg, port);
+  }
+
+  addServer(sg: ec2.ISecurityGroup, port: ec2.Port) {
+    this.sg.addEgressRule(sg, port);
+    sg.addIngressRule(this.sg, port);
+  }
+}
+
+export class LockerSetup {
+  locker_ec2: LockerEc2;
+  db_cluster: DatabaseCluster;
+  db_sg: SecurityGroup;
+  db_bucket: s3.Bucket;
+
+  constructor(scope: Construct, vpc: ec2.Vpc, config: LockerConfig) {
+    // Creating Database for LockerData
+    const engine = DatabaseClusterEngine.auroraPostgres({
+      version: AuroraPostgresEngineVersion.VER_13_7,
+    });
+
+    const db_name = "locker";
+
+    const db_security_group = new SecurityGroup(scope, "Locker-db-SG", {
+      securityGroupName: "Locker-db-SG",
+      vpc: vpc,
+    });
+
+    this.db_sg = db_security_group;
+
+    const secretName = "locker-db-master-user-secret";
+
+    // Create the secret if it doesn't exist
+    let secret = new Secret(scope, "locker-db-master-user-secret", {
```

Comment:
will this secret throw error for `secret already present` when we run it second time? 
If it does then pls fix it to work with mulitple updates